traitsui/qt4/button_editor.py

```diff
@@ -63,11 +63,8 @@ def init(self, parent):
             self.control = QtGui.QToolButton()
             self.control.toolButtonStyle = QtCore.Qt.ToolButtonTextOnly
             self.control.setText(self.string_value(label))
-            self.object.on_trait_change(
-                self._update_menu, self.factory.values_trait
-            )
-            self.object.on_trait_change(
-                self._update_menu, self.factory.values_trait + "_items"
+            self.object.observe(
+                self._update_menu, self.factory.values_trait + ".items"
```

Comment:
IIUC this works because for `container` traits, we can simply use `container.items` to listen to both changes to the `container` trait and the `items` in the container.